### PR TITLE
refactor(auth): removed fxa-auth-server/fxa-oauth-server

### PR DESCRIPTION
**Pull request by dannycoates**
Holding for train-151

This deletes the fxa-oauth-server and removes it from CI

### CODE REVIEW

File: packages/fxa-auth-server/fxa-oauth-server/scripts/generate-client-for-ops.js [hunk 1]
```diff
@@ -1,54 +0,0 @@
-#!/usr/bin/env node
-
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-if (process.argv.length !== 4) {
-  console.error(
```
Comment (shane-tomlinson) [inline packages/fxa-auth-server/fxa-oauth-server/scripts/generate-client-for-ops.js]: Has this moved elsewhere? We use this pretty regularly when creating new clients.

File: packages/fxa-auth-server/fxa-oauth-server/scripts/gen_keys.js [hunk 2]
```diff
@@ -1,89 +0,0 @@
-#!/usr/bin/env node
-
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-/*
-
- Usage:
```
Comment (shane-tomlinson) [inline packages/fxa-auth-server/fxa-oauth-server/scripts/gen_keys.js]: No longer needed?

File: packages/fxa-auth-server/fxa-oauth-server/scripts/prepare-new-signing-key.js [hunk 3]
```diff
@@ -1,49 +0,0 @@
-#!/usr/bin/env node
-
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-/*
```
Comment (shane-tomlinson) [inline packages/fxa-auth-server/fxa-oauth-server/scripts/prepare-new-signing-key.js]: I believe @jrgm and @jbuck use prepare-new-signing-key, retire-old-signing-key, and sops-key-config to manage the OIDC and JWT access token signing keys, we'll need those still.

File: packages/fxa-auth-server/fxa-oauth-server/scripts/rpm-version.js [hunk 4]
```diff
@@ -1,37 +0,0 @@
-#!/usr/bin/env node
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-var cp = require('child_process');
-var util = require('util');
-
-// Generate legacy-format output that looks something like:
```
Comment (shane-tomlinson) [inline packages/fxa-auth-server/fxa-oauth-server/scripts/rpm-version.js]: @jrgm - do you use this?

**Comment by shane-tomlinson** [comment]
Screenshot 2019-11-12 at 12 40 49

🎆

File: packages/fxa-auth-server/fxa-oauth-server/scripts/gen_keys.js (hunk 2 above)
Comment (dannycoates) [reply packages/fxa-auth-server/fxa-oauth-server/scripts/gen_keys.js]: `fxa-auth-server/scripts/oauth_gen_keys.js`

File: packages/fxa-auth-server/fxa-oauth-server/scripts/prepare-new-signing-key.js (hunk 3 above)
Comment (dannycoates) [reply packages/fxa-auth-server/fxa-oauth-server/scripts/prepare-new-signing-key.js]: in `fxa-auth-server/scripts`

File: packages/fxa-auth-server/fxa-oauth-server/scripts/rpm-version.js (hunk 4 above)
Comment (dannycoates) [reply packages/fxa-auth-server/fxa-oauth-server/scripts/rpm-version.js]: `fxa-auth-server/scripts`

File: packages/fxa-auth-server/fxa-oauth-server/bin/purge_expired_tokens.js [hunk 5]
```diff
@@ -1,138 +0,0 @@
-#!/usr/bin/env node
-
-/* This Source Code Form is subject to the terms of the Mozilla Public
```
[the remark this answers is not in the document]
Comment (dannycoates) [reply packages/fxa-auth-server/fxa-oauth-server/bin/purge_expired_tokens.js]: moved

File: packages/fxa-auth-server/fxa-oauth-server/docs/clients.md [hunk 6]
```diff
@@ -1,65 +0,0 @@
-# Configuring OAuth clients
```
[the remark this answers is not in the document]
Comment (dannycoates) [reply packages/fxa-auth-server/fxa-oauth-server/docs/clients.md]: moved to docs/oauth

File: packages/fxa-auth-server/fxa-oauth-server/scripts/generate-client-for-ops.js (hunk 1 above)
Comment (dannycoates) [reply packages/fxa-auth-server/fxa-oauth-server/scripts/generate-client-for-ops.js]: moved to fxa-auth-server/scripts

**Comment by shane-tomlinson** [comment]
@dannycoates - are there plans to remove the [service->service calls](https://github.com/mozilla/fxa/blob/master/packages/fxa-auth-server/fxa-oauth-server/lib/auth_server.js#L11) and make direct DB calls instead? I imagine we'll be able to shed significant code complexity as well as speed up response time since we'd no longer need to do, e.g., auth->oauth->auth calls when creating JWT access tokens.

**Comment by dannycoates** [comment]
> are there plans to remove the [service->service calls](https://github.com/mozilla/fxa/blob/master/packages/fxa-auth-server/fxa-oauth-server/lib/auth_server.js#L11) and make direct DB calls instead?

most definitely! The first to go was the oauth->auth call in [generateAccessToken](https://github.com/mozilla/fxa/blob/master/packages/fxa-auth-server/lib/oauth/grant.js#L242)

**Pull request merged**